File: docs/snippets/all/archetypes/points3d_random.rs
//! Log some random points with color and radii.

use rand::prelude::*;

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let rec = rerun::RecordingStreamBuilder::new("rerun_example_points3d_random").spawn()?;

    let mut rng = rand::rngs::SmallRng::seed_from_u64(42);
    let dist = rand::distr::Uniform::new(-5., 5.)?;

    rec.log(
        "random",
        &rerun::Points3D::new(
            (0..10).map(|_| (rng.sample(dist), rng.sample(dist), rng.sample(dist))),
        )
        .with_colors(
            (0..10).map(|_| rerun::Color::from_rgb(rng.random(), rng.random(), rng.random())),
        )
        .with_radii((0..10).map(|_| rng.random::<f32>())),
    )?;

    Ok(())
}

File: docs/snippets/all/archetypes/points3d_random.py
"""Log some random points with color and radii."""

from numpy.random import default_rng

import rerun as rr

rr.init("rerun_example_points3d_random", spawn=True)
rng = default_rng(12345)

positions = rng.uniform(-5, 5, size=[10, 3])
colors = rng.uniform(0, 255, size=[10, 3])
radii = rng.uniform(0, 1, size=[10])

rr.log("random", rr.Points3D(positions, colors=colors, radii=radii))
